Classes/Domain/Repository/CommentRepository.php

```diff
@@ -246,12 +246,23 @@ protected function getFindByEmailAndPostIdConstraints(QueryInterface $query, $em
      *
      * @return object
      */
-    protected function getValidConstraints(QueryInterface $query)
+    protected function getValidConstraints(QueryInterface $query, $excludeHiddenPosts = false)
     {
         $constraints = $query->logicalAnd(
             $query->equals('spam', 0),
             $query->equals('approved', 1)
         );
+        // By default, comments of hidden posts will be included in the result, because there is no relation / join from comments to posts
+        // To exclude these comments, we join w/ post (via post.uid constraint)
+        // XXX Is there another / better way to join, w/o a unneeded constraint?
+        // The generated query automatically excludes hidden posts via default enablefields handling
+        // (the generated SQL contains ... tx_t3blog_post.deleted = 0 ... AND tx_t3blog_post.hidden = 0 ...)
+        if ($excludeHiddenPosts) {
+            $constraints = $query->logicalAnd(
+                $constraints,
+                $query->greaterThan('post.uid', 0)
```

Comment:
When not using an `ObjectStorage` but a simple relation you could just use `$query->equals('post.hidden', 0)`. Anyway, this will not help to retrieve hidden posts as the internal data mapper does not use the repository (and it's behavior regarding hidden or deleted relations is not configurable).